### ISSUE TITLE
extraTokenParams not used in signinSilent()

### ISSUE THREAD

**Issue by SorenKrogh** [authts/oidc-client-ts#405]
After signing in through KeyCloak I would like to use signinSilent() with extraTokenParams to set additional roles for the user.
signinSilent however does not pass on the extraTokenParams params to KeyCloak.
If I use signinRedirect() or signinPopup() the extraTokenParams are passed on to KeyCloak and the additional roles are correctly added to the new access_token.
In my use case I much prefer to use the signinSilent() to assign additional roles to an already signed in user.
Is this possible, or should it be possible?

**Comment by pamapa** [comment]
`signinSilent` supports as parameter optionally `SigninSilentArgs`. `type SigninSilentArgs = IFrameWindowParams & ExtraSigninRequestArgs`. For cases without a refresh token (prompt=none) `extraTokenParams` and `extraQueryParams` should be passed via `requestArgs` see https://github.com/authts/oidc-client-ts/blob/0642515bf700ea071f1b136be250a1c950f9de90/src/UserManager.ts#L251.

`signinSilent` is thought to refresh the access token. To enrich the user profile have a look at the setting `loadUserInfo`.

**Comment by SorenKrogh** [comment]
In my case I do have a refresh token. However, I still need to pass two extra parameters in `extraTokenParams` (or any other way to pass parameters in the payload of the refresh request). If these parameters aren't passed when refreshing the token, the assigned roels will not be set in the refreshed token.
The roles are embedded in the token to allow the client to fetch restricted data from the backend.
In short, the refresh token isn't enough to preserve the obtained roles when refreshing the access token with `signinSilent`.

**Comment by pamapa** [comment]
When i read the _OAuth 2.0 Authorization_ [spec](https://datatracker.ietf.org/doc/html/rfc6749#section-6), i do **not** see that in case of _refreshing an access token_ the `extraTokenParams` are passed. You can pass optionally `scope` after the spec.

**Comment by pamapa** [comment]
unless you disagree and comment why, i will close this issue soon
